tests/cli.py

```diff
@@ -785,5 +785,27 @@ def test_ip_leases_no_nm_lease(self):
         self.assertIn(b'No lease found', err)
         self.assertNotEqual(p.returncode, 0)
 
+    def test_ip_leases_generate_mapping_is_empty(self):
+        mock_generate = MockCmd('generate')
```

Comment:
There's a MockCmd.set_returncode() method, which should be used here, to mimick `netplan ip`'s behavior.